### ISSUE TITLE
Fix FreshNameGenerator

### ISSUE THREAD

**Issue by cacay** [apl-cornell/viaduct#61]
The fresh name generator does not take numbers in the base name into account. For example, the program
```Kotlin
process main {
    {
        val tmp: int = 0;
    }
    {
        val tmp: int = 0;
    }
    val tmp_1: int = 0;
}
```
elaborates to
```Kotlin
process main {
    {
        val tmp: int = 0;
    }
    {
        val tmp_1: int = 0;
    }
    val tmp_1: int = 0;
}
```

Note that there are two `tmp_1`s. 

One solution is to parse base names in `FreshNameGenerator` to strip the `_XXX` component. Another is to repeatedly call `FreshNameGenerator` until we find a unique name Cuckoo hashing style.

See 98a4e451c32d0791c43e184ad83021c4e055932f.

**Comment by rolph-recto** [comment]
closed by https://github.com/apl-cornell/viaduct/pull/74

**Comment by cacay** [comment]
It's not fixed: #76 

Here's the essence of the problem:
https://github.com/apl-cornell/viaduct/blob/02f81ef03bba86913416155026b5ee772e19a242/compiler/src/test/kotlin/edu/cornell/cs/apl/viaduct/util/FreshNameGeneratorTest.kt#L52-L59
